##1.	Suppose we want to test whether the mean weight of apples in a grocery store is 150 grams. We randomly sample 20 apples from the store and measure their weights, getting the following data:

In [9]:
Apple_weights = [145, 155, 160, 146, 142, 152, 150, 147, 148, 149, 148, 152, 153, 155, 154, 148, 151, 147, 153, 146]
Apple_weights

[145,
 155,
 160,
 146,
 142,
 152,
 150,
 147,
 148,
 149,
 148,
 152,
 153,
 155,
 154,
 148,
 151,
 147,
 153,
 146]

##	What test should we use and why?

-One-sample t-test to determine the mean weight of the apples
- This test is used because here we are given sample size and also the population standard deviation is unknown.

##	State the null and alternative hypotheses.


-H0:population mean weight of the apples is equal to 150 gms

-Ha:Population mean weigth of the apples is not equal to 150 gms 

##	Choose a significance level (α) (the probability of rejecting the null hypothesis when it is actually true).

alpha =0.05

##Determine the degrees of freedom (df) of the sample. Df = sample size -1 

Here our sample size is 20, so degree of the freedom is calculated as 
df = 20-1
df =19
 so the degree of the freedom for our sampl is 19 

##Determine the critical value of t based on the significance level and degrees of freedom. For a two-tailed test with α = 0.05 and df = 19, the critical values are -2.093 and 2.093.

In [10]:
import numpy as np
from scipy import stats

# Define the sample data
Apple_weights = [145, 155, 160, 146, 142, 152, 150, 147, 148, 149, 148, 152, 153, 155, 154, 148, 151, 147, 153, 146]

# Define the null hypothesis mean
mu = 150

# Calculate the sample mean and standard deviation
sample_mean = np.mean(Apple_weights)
sample_std = np.std(Apple_weights, ddof=1)

# Calculate the t-statistic and p-value
t_stat = (sample_mean - mu) / (sample_std / np.sqrt(len(Apple_weights)))
#Calculate the  p-value
p_value = 2 * stats.t.sf(np.abs(t_stat), df=len(Apple_weights)-1)

# Print the results
print("Sample Mean:",sample_mean)
print("Sample Std:",sample_std)
print("t-statistic = ", round(t_stat,3))
print("p-value = ", round(p_value,5))

# Compare the t-statistic to the critical values at alpha = 0.05 and df = 19
alpha = 0.05
t_crit_lower = -2.093
t_crit_upper = 2.093

if t_stat < t_crit_lower or t_stat > t_crit_upper:
    print("Reject the null hypothesis")
else:
    print("Fail to reject the null hypothesis")

Sample Mean: 150.05
Sample Std: 4.2608993367474355
t-statistic =  0.052
p-value =  0.9587
Fail to reject the null hypothesis


So ,we can see that based on the values on the t-statistics and also the critical values of two tailed test with alpha value of 0.05 and df = 19 results to Fail to reject the null hypothesis which means that the mean of the weight of the apples = 150grams

##	Compare and interpret the results of the test to the critical value 

Based upon the results we could see that the t-statistic value we obtained is 0.052 ,it within the critical value of -2.093 and 2.093 ,hence we could say that we fail to reject the null hypothesis at 5% significan level.
We can also see that the P-value is greater than 0.05,so we fail to reject the null the hypothsis.

##2.	Suppose we want to test whether the mean height of all men in a population is 180 cm assuming that the population standard deviation = 2, We randomly sample 50 men from the population and measure their heights.

In [11]:
Men_height = [177, 180, 182, 179, 178, 181, 176, 183, 179, 180, 178, 181, 177, 178, 180, 179, 182, 180, 183, 181, 179, 177, 180, 181, 178, 180, 182, 179, 177, 182, 178, 181, 183, 179, 180, 181, 183, 178, 177, 181, 179, 182, 180, 181, 178, 180, 179, 181, 183, 179]
Men_height

[177,
 180,
 182,
 179,
 178,
 181,
 176,
 183,
 179,
 180,
 178,
 181,
 177,
 178,
 180,
 179,
 182,
 180,
 183,
 181,
 179,
 177,
 180,
 181,
 178,
 180,
 182,
 179,
 177,
 182,
 178,
 181,
 183,
 179,
 180,
 181,
 183,
 178,
 177,
 181,
 179,
 182,
 180,
 181,
 178,
 180,
 179,
 181,
 183,
 179]

## What test should we use and why?

We will choose one sample Z test, because as we have the population standard mean = 2 and the sample size is large n= 50

##State the null and alternative hypotheses.

 - H0: mean height of all men is 180 cm

 - Ha:  mean height of all men is not 180cm (two-tailed)

 - Ha:   mean height of all men is either less than or greater than 180cm (one-tailed)

##	Choose a significance level (α) (the probability of rejecting the null hypothesis when it is actually true).

alpha =0.05

## Determine the degrees of freedom (df) of the sample. Df = sample size -1 

Here our sample size is 50, so degree of the freedom is calculated as df = 50-1 df =49 so the degree of the freedom for our sampl is 49

##Determine the critical value of t based on the significance level and degrees of freedom. For a two-tailed test with α = 0.05 and df = 19, the critical values are -2.093 and 2.093

In [12]:
from scipy.stats.distributions import alpha_gen
from numpy.lib import math
import numpy as np
import math

from numpy.random import randn
from statsmodels.stats.weightstats import ztest as ztest


Men_height = [177, 180, 182, 179, 178, 181, 176, 183, 179, 180, 178, 181, 177, 178, 180, 179, 182, 180, 183, 181, 179, 177, 180, 181, 178, 180, 182, 179, 177, 182, 178, 181, 183, 179, 
               180, 181, 183, 178, 177, 181, 179, 182, 180, 181, 178, 180, 179, 181, 183, 179]

#sample_mean=np.array(Men_height).mean()
#sample_std=2/math.sqrt(50)
alpha =0.05
pop_mean =180

print("The sample mean is ",sample_mean)
    
ztest_Score, p_value= ztest(Men_height,value = pop_mean, alternative='larger')# passing our sample data,mean, Ha to check if the mean is larger

print("z testscore",round(ztest_Score,4),"P Value ",round(p_value,4)) # coresponding Z score and P value
print("alpha value ",alpha)
# Compare if p_value  is greater thhan alpha accept null hypothesis else reject
if(p_value <  alpha):
  print("Reject Null Hypothesis")
else:
  print("Fail to Reject NUll Hypothesis")

The sample mean is  150.05
z testscore -0.6061 P Value  0.7278
alpha value  0.05
Fail to Reject NUll Hypothesis


##Compare and interpret the results of the test to the critical value

As we can see the p_value is greater than the 0.05,so we have significant evidence ,which means we Fail to reject the null hypothesis.So in other words we can say that mean of the height is  equal to 180

##3.	Suppose we want to test whether the mean weight of a population of cats is different from 4 kg. We randomly sample 50 cats from the population and measure their weights, getting the following data:

Cats_weights = [3.9, 4.2, 4.5, 4.1, 4.3, 3.8, 4.6, 4.2, 3.7, 4.3, 3.9, 4.0, 4.1, 4.5, 4.2, 3.8, 3.9, 4.3, 4.1, 4.0, 4.4, 4.2, 4.1, 4.6, 4.4, 4.2, 4.1, 4.3, 4.0, 4.4, 4.3, 3.8, 4.1, 4.5, 4.2, 4.3, 4.0, 4.1, 4.2, 3.9, 4.3, 3.7, 4.1, 4.5, 4.2, 4.0, 4.2, 4.4, 4.1, 4.5]



In [13]:
from scipy.stats.distributions import alpha_gen
from numpy.lib import math
import numpy as np
import math

from statsmodels.stats.weightstats import ztest 

Cats_weights = [3.9, 4.2, 4.5, 4.1, 4.3, 3.8, 4.6, 4.2, 3.7, 4.3, 3.9, 4.0, 4.1, 4.5, 4.2, 3.8, 3.9, 4.3, 4.1, 4.0, 4.4, 4.2, 4.1, 4.6, 4.4, 4.2, 4.1, 4.3, 4.0, 4.4, 4.3, 3.8, 4.1, 4.5, 
                4.2, 4.3, 4.0, 4.1, 4.2, 3.9, 4.3, 3.7, 4.1, 4.5, 4.2, 4.0, 4.2, 4.4, 4.1, 4.5]

#population mean given
mu=4

#sample size 50

#mean of the sample
samp_cat_weight_mean=np.array(Cats_weights).mean()

alpha =0.05
pop_mean = 4

print("The sample mean is ",round(samp_cat_weight_mean,1))

alpha =0.05
pop_mean =4

ztest_Score, p_value= ztest(Cats_weights,value = pop_mean, alternative='two-sided')# compute the Z score and P value


print("z score",round(ztest_Score,4),"P Value ",round(p_value,4))
#comapre p value with the alpha value if its less than we reject the null hypothesis else accept the null hypothesis
if(p_value <  alpha):
  print("Reject Null Hypothesis")
else:
  print("Fail to Reject NUll Hypothesis")


The sample mean is  4.2
z score 5.2336 P Value  0.0
Reject Null Hypothesis


- From the result we can say that the mean weigth of the cats is significantly different from 4kg.
- So we reject the null hypothesis,thus we could suggest that the population mean weight of the cats is not equal to 4 kg.
- P- value = 0.0 so we can say that there is significantly difference between the sample mean and hypothesised population mean.

## State the null and alternative hypotheses.

- H0: The mean weight of the population of cats is equal to 4kg
- Ha : The mean weight of the population of cats is not equal to 4 kg

## Choose a significance level (α) (the probability of rejecting the null hypothesis when it is actually true).

-Significance level ,alpha = 0.05

## ○	Calculate the z-score using the formula:

In [14]:
import math

Cats_weights = [3.9, 4.2, 4.5, 4.1, 4.3, 3.8, 4.6, 4.2, 3.7, 4.3, 3.9, 4.0, 4.1, 4.5, 4.2, 3.8, 3.9, 4.3, 4.1, 4.0, 4.4, 4.2, 4.1, 4.6, 4.4, 4.2, 4.1, 4.3, 4.0, 4.4, 4.3, 3.8, 4.1, 4.5, 4.2, 4.3, 4.0, 4.1, 4.2, 3.9, 4.3, 3.7, 4.1, 4.5, 4.2, 4.0, 4.2, 4.4, 4.1, 4.5]

sample_mean = sum(Cats_weights)/len(Cats_weights) # sample mean
pop_mean = 4
pop_std = sample_mean / math.sqrt(len(Cats_weights)) # population standard deviation

z_score = (sample_mean - pop_mean) / pop_std # calculating z score with the formula

print("Sample Mean:",round(sample_mean,3))
print("Population Standard Deviation:",round(pop_std,4))
print("The z-score is:", round(z_score,4))
if(z_score > 1.96 or z_score< -1.96):
    print("Reject the null hypothesis")
else:
    print("Fail to Reject NUll Hypothesis")



Sample Mean: 4.17
Population Standard Deviation: 0.5897
The z-score is: 0.2883
Fail to Reject NUll Hypothesis


## ○	Assuming that the standard deviation is equal to the sample mean

In [16]:
import math

Cats_weights = [3.9, 4.2, 4.5, 4.1, 4.3, 3.8, 4.6, 4.2, 3.7, 4.3, 3.9, 4.0, 4.1, 4.5, 4.2, 3.8, 3.9, 4.3, 4.1, 4.0, 4.4, 4.2, 4.1, 4.6, 4.4, 4.2, 4.1, 4.3, 4.0, 4.4, 4.3, 3.8, 4.1, 4.5, 4.2, 4.3, 4.0, 4.1, 4.2, 3.9, 4.3, 3.7, 4.1, 4.5, 4.2, 4.0, 4.2, 4.4, 4.1, 4.5]

sample_mean = sum(Cats_weights)/len(Cats_weights) # Sample mean
pop_mean = 4 # population mean
pop_std = sample_mean # Assuming that the standard deviation is equal to the sample mean
# Sample size
n = len(Cats_weights)
# Calculate the z-score
z_score = (sample_mean - pop_mean)/(pop_std/math.sqrt(n))

print("Sample Mean:",round(sample_mean,3))
print("Population Standard Deviation:",round(pop_std,4))
print("The z-score is:", round(z_score,4))
if(z_score > 1.96 or z_score< -1.96):
    print("Reject the null hypothesis")
else:
    print("Fail to Reject NUll Hypothesis")


Sample Mean: 4.17
Population Standard Deviation: 4.17
The z-score is: 0.2883
Fail to Reject NUll Hypothesis


## ○	Look up the critical z-value at the chosen significance level (α) using a z-table.

- We have the alpha value = 0.05, so from the z-table  we find that critical z values are +/- 1.96
- if ou z score tends to lies outside this range , it result to reject our null hypothesis else we accept our null hypothesis.

## ○	Compare the calculated z-score to the critical z-values. If the calculated z-score falls outside the range between the critical z-values, we reject the null hypothesis in favor of the alternative hypothesis.

The result of our z testscore 0.2883,  the critical value we observed from the table is +/- 1.96, so the value lies outside the range .Hence we  Fail to reject the null hypothesis.
Therefore, we can say that there is no significantly difference in the mean weight of the cat from 4kg.